frontend/src/pages/Graphing/Combobox/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { ComboboxSelect, Text } from '@highlight-run/ui/components'
+import _ from 'lodash'
+import moment from 'moment'
+import React, { useEffect, useMemo, useState } from 'react'
+import { useDebounce } from 'react-use'
+
+import { TIME_FORMAT } from '@/components/Search/SearchForm/constants'
+import { useGetKeysLazyQuery } from '@/graph/generated/hooks'
+import { KeyType, ProductType } from '@/graph/generated/schemas'
+import { useProjectId } from '@/hooks/useProjectId'
+
+import * as style from './styles.css'
+
+type Props = {
+	selection: string
+	setSelection: (selection: string) => void
+	label: string
+	searchConfig: {
+		productType: ProductType
+		startDate: Date
+		endDate: Date
+	}
+	onlyNumericKeys?: boolean
+	defaultKeys?: string[]
+}
+
+export const Combobox: React.FC<Props> = ({
+	selection,
+	setSelection,
+	label,
+	searchConfig: { productType, startDate, endDate },
+	onlyNumericKeys,
+	defaultKeys,
+}) => {
+	const { projectId } = useProjectId()
+	const [getKeys, { data }] = useGetKeysLazyQuery()
+	const [query, setQuery] = useState('')
+	const [debouncedQuery, setDebouncedQuery] = useState('')
+	useDebounce(
+		() => {
+			setDebouncedQuery(query)
+		},
+		300,
+		[query],
+	)
+
+	const keyOptions = useMemo(() => {
+		const baseKeys: string[] = []
+
+		for (const key of defaultKeys || []) {
+			if (key.toLowerCase().includes(query.toLowerCase())) {
+				baseKeys.push(key)
+			}
+		}
+
+		const searchKeys =
+			_.chain(data?.keys || [])
+				.map('name')
+				.uniq()
+				.value() ?? []
+
+		baseKeys.concat(searchKeys).slice(0, 8)
+
+		return baseKeys.concat(searchKeys).map((o) => ({
+			key: o,
+			render: o,
+		}))
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [data?.keys, defaultKeys])
+
+	useEffect(() => {
+		getKeys({
```

Comment:
Moves this into the combobox component so each combobox can decide whether to use numeric or all keys